src/compile.rs

```diff
@@ -1049,13 +1055,6 @@ impl ByteClassSet {
     }
 }
 
-fn u32_to_usize(n: u32) -> usize {
```

Comment:
Actually, I think this is the only really useful warning by clippy for the whole codebase: The comparison always evaluates to false, making the check superfluous.

Why is it there anyway? if `usize` is 32bit, `n` can't be larger. If `usize` is 64bit, `n` can't be larger as well.